server/src/main/java/org/spine3/server/command/error/CommandException.java

```diff
@@ -37,30 +38,52 @@
     private final Command command;
     private final Error error;
 
+    /**
+     * Creates a new instance.
+     *
+     * @param messageText an error message text
+     * @param command a related command
+     * @param error an error occurred
+     */
     public CommandException(String messageText, Command command, Error error) {
         super(messageText);
-        this.error = error;
         this.command = command;
+        this.error = error;
     }
 
-    public static Map<String, Value> commandTypeAttribute(String commandType) {
-        return ImmutableMap.of(Attribute.COMMAND_TYPE_NAME, Value.newBuilder().setStringValue(commandType).build());
+    /**
+     * Returns a map with a command type attribute.
+     *
+     * @param commandMessage a command message to get the type from
+     */
+    public static Map<String, Value> commandTypeAttribute(Message commandMessage) {
+        final String commandType = commandMessage.getDescriptorForType().getFullName();
```

Comment:
How about using `TypeName` for getting the type name string? It would be a bit more compact.